src/main/java/com/epam/esm/dao/impl/TagDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package com.epam.esm.dao.impl;
+
+import com.epam.esm.dao.DaoException;
+import com.epam.esm.dao.TagDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import com.epam.esm.model.Tag;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.MultiValueMap;
+
+import java.util.List;
+
+import static com.epam.esm.dao.SqlProvider.*;
+
+
+@Repository
+@Transactional(readOnly = true)
+public class TagDaoImpl implements TagDao {
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    @Override
+    public List<Tag> getAll() {
+        return jdbcTemplate.query(GET_ALL_TAGS, (rs, rowNum) -> {
+            Tag tag = new Tag();
+            tag.setId(rs.getLong(1));
+            tag.setName(rs.getString(2));
+            return tag;
+        });
+    }
+
+    @Override
+    public List<Tag> getByParameters(MultiValueMap<String, String> params) throws DaoException {
+        return null;
```

Comment:
return exception (for example UnsupportedOperationException) that it's not supported

Reply:
Method reworked, now it returns jdbcTemplate.query()